### PR TITLE
Add a developer mode

### LINKED ISSUE #16: Dev endpoint capability
Filing an issue at suggestion of @knowtheory . It would be useful for development purposes to be able to redirect data pings to a local endpoint. This would allow development work to commence prior to signing of a data use agreement and granting of access to the Mozilla environment.

### LINKED ISSUE #35: invalid id error while using sendPing
Error details: background.js:1418 Rally.sendPing - error while sending geolocation TypeError: Error in invocation of runtime.sendMessage(optional string extensionId, any message, optional object options, optional function responseCallback): Invalid extension id: 'rally-core@mozilla.org' 

PS: I'm using the default keys that are already there in the code to initialize rally.

### PR DESCRIPTION
**Pull request by Dexterp37**
This fixes #16, fixes #35 

This PR updates the partner support library to version 0.1.0, enabling the use of the _developer mode_. In this mode:

- Pings are dumped to the browser console instead of being sent (to aid debug).
- The webpage for the Rally website is not opened if the core-addon is not detected.

cc @jcchuang

### CODE REVIEW

**Comment by Dexterp37** [comment]
@rhelmer looks like I'm stumbling on a Rollup.js bug locally on my machine, and I'm not able to make it watch. Any chance you could try this? Either with `npm run watch` (and making a change to `study.js` then wait for the browser to reload the addon) or `npm run watch-addon`  (and making a change to `study.js` then wait for `dist/background.js` to change).

**Comment by rhelmer** [comment]
> @rhelmer looks like I'm stumbling on a Rollup.js bug locally on my machine, and I'm not able to make it watch. Any chance you could try this? Either with `npm run watch` (and making a change to `study.js` then wait for the browser to reload the addon) or `npm run watch-addon` (and making a change to `study.js` then wait for `dist/background.js` to change).

Yeah sorry actually the reason I hadn't approved this yet is because it didn't seem to be working :) I'll give it a shot.

**Comment by rhelmer** [comment]
> @rhelmer looks like I'm stumbling on a Rollup.js bug locally on my machine, and I'm not able to make it watch. Any chance you could try this? Either with `npm run watch` (and making a change to `study.js` then wait for the browser to reload the addon) or `npm run watch-addon` (and making a change to `study.js` then wait for `dist/background.js` to change).

OK yes this is not working for me locally either, I can help take a look in my morning.

File: package.json [hunk 1]
```diff
@@ -3,25 +3,31 @@
...
     "build": "rollup -c",
+    "dev": "rollup -c --config-enable-developer-mode",
     "lint": "eslint . && web-ext lint",
     "package": "npm run build && web-ext build --overwrite-dest --ignore-files \"./tests/**/*\" --ignore-files \"./src/**/*\" --ignore-files \"./rollup.config.js\"",
     "start": "npm run build && web-ext run",
-    "test-integration": "npm run package && mv web-ext-artifacts/*.zip web-ext-artifacts/study.xpi && mocha --timeout 30000 \"./tests/integration/*.js\""
+    "test-integration": "npm run package && mv web-ext-artifacts/*.zip web-ext-artifacts/study.xpi && mocha --timeout 30000 \"./tests/integration/*.js\"",
+    "watch": "npm-run-all --parallel watch-addon watch-webext",
+    "watch-addon": "npm run dev -w",
+    "watch-webext": "web-ext run --watch-file dist/background.js"
```
Comment (rhelmer) [inline package.json:13]: Nit, but I am confused about the distinction between `webext` and `addon` here :) The usual way these terms are used is that a Add-ons are an umbrella term that encompasses both WebExtensions and Themes (and formerly legacy Plug-ins, these are all deprecated except for the DRM stuff but you can still see them in `about:addons`)

Comment (Dexterp37) [reply package.json]: Thanks for the clarification. I renamed the scripts. Is this better?

**Comment by Dexterp37** [comment]
@rhelmer looks like updating the dependencies and correctly passing along `-w` did it :-)

**Review by rhelmer — APPROVED** [review]
Thanks for doing this! I'm really looking forward to using it.

**Pull request merged**